library/std/src/sys/pal/uefi/process.rs

```diff
@@ -0,0 +1,712 @@
+pub enum Stdio {
+    Inherit,
+    Null,
+    MakePipe,
+}
+
+impl Command {
+    pub fn new(program: &OsStr) -> Command {
+        Command { prog: program.to_os_string(), args: Vec::new(), stdout: None, stderr: None }
+    }
+
+    pub fn arg(&mut self, arg: &OsStr) {
+        self.args.push(arg.to_os_string());
+    }
+
+    pub fn env_mut(&mut self) -> &mut CommandEnv {
+        panic!("unsupported")
+    }
+
+    pub fn cwd(&mut self, _dir: &OsStr) {
+        panic!("unsupported")
+    }
+
+    pub fn stdin(&mut self, _stdin: Stdio) {
+        panic!("unsupported")
+    }
+
+    pub fn stdout(&mut self, stdout: Stdio) {
+        self.stdout = Some(stdout);
+    }
+
+    pub fn stderr(&mut self, stderr: Stdio) {
+        self.stderr = Some(stderr);
+    }
+
+    pub fn get_program(&self) -> &OsStr {
+        self.prog.as_ref()
+    }
+
+    pub fn get_args(&self) -> CommandArgs<'_> {
+        CommandArgs { iter: self.args.iter() }
+    }
+
+    pub fn get_envs(&self) -> CommandEnvs<'_> {
+        panic!("unsupported")
+    }
+
+    pub fn get_current_dir(&self) -> Option<&Path> {
+        None
+    }
+
+    pub fn spawn(
+        &mut self,
+        _default: Stdio,
+        _needs_stdin: bool,
+    ) -> io::Result<(Process, StdioPipes)> {
+        unsupported()
+    }
+
+    fn create_pipe(
+        s: Stdio,
+    ) -> io::Result<Option<helpers::OwnedProtocol<uefi_command_internal::PipeProtocol>>> {
+        match s {
+            Stdio::MakePipe => helpers::OwnedProtocol::create(
+                uefi_command_internal::PipeProtocol::new(),
+                simple_text_output::PROTOCOL_GUID,
+            )
+            .map(Some),
+            Stdio::Null => helpers::OwnedProtocol::create(
+                uefi_command_internal::PipeProtocol::null(),
+                simple_text_output::PROTOCOL_GUID,
+            )
+            .map(Some),
+            Stdio::Inherit => Ok(None),
+        }
+    }
+
+    pub fn output(&mut self) -> io::Result<(ExitStatus, Vec<u8>, Vec<u8>)> {
+        let mut cmd = uefi_command_internal::Image::load_image(&self.prog)?;
+
+        /* Setup Stdout */
+        let stdout = self.stdout.unwrap_or(Stdio::MakePipe);
+        let stdout = Self::create_pipe(stdout)?;
+        if let Some(con) = stdout {
+            cmd.stdout_init(con)
+        } else {
+            cmd.stdout_inherit()
+        };
+
+        /* Setup Stderr */
+        let stderr = self.stderr.unwrap_or(Stdio::MakePipe);
+        let stderr = Self::create_pipe(stderr)?;
+        if let Some(con) = stderr {
+            cmd.stderr_init(con)
+        } else {
+            cmd.stderr_inherit()
+        };
+
+        // No reason to set args if only program name is preset
+        if !self.args.is_empty() {
+            let args = self.args.iter().fold(OsString::from(&self.prog), |mut acc, arg| {
+                acc.push(" ");
+                acc.push(arg);
+                acc
+            });
+            cmd.set_args(&args);
```

Comment:
Do you mean I should extract argument parsing to it's own function and leave the current logic, or just have a blank implementation for handling arguments for now?

Also, what are the rules regarding args?
1. Is the user supposed to put `"` around string arguments (those that contain whitespace). Or is the implementation supposed to detect if quote is needed?
2. Is the user supposed to escape the characters using `^` (UEFI escape), or should the implementation detect characters that need to be escaped?
3. What happens if user passes a NULL character?

Reply:
We need to do the escaping so that `env::args()` in the image can recover the exact arguments passed in here. As for NULL, UNIX remembers whether one of the arguments contains NULL inside the `Command` structure and returns an error upon spawning in that case.